athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsRecordHandler.java

```diff
@@ -167,17 +169,23 @@ protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recor
      * We are writing data to spiller. This function received the whole batch
      * along with row index. We will access into batch using the row index and
      * get the record to write into spiller.
-     *
-     * @param spiller         - block spiller
+     *  @param spiller         - block spiller
      * @param gcsFieldVectors - the batch
      * @param rowIndex        - row index
+     * @param partitionColumns   - partition column
+     * @param split           - split
      */
     private void execute(
             BlockSpiller spiller,
-            List<FieldVector> gcsFieldVectors, int rowIndex)
+            List<FieldVector> gcsFieldVectors, int rowIndex, List<Field> partitionColumns, Split split)
     {
         spiller.writeRows((Block block, int rowNum) -> {
             boolean isMatched = true;
+            // offer value for partition column
+            for (Field field : partitionColumns) {
+                isMatched &= block.offerValue(field.getName().toLowerCase(), rowNum, split.getProperty(field.getName().toLowerCase()));
+            }
```

Comment:
Are there already tests checking to make sure that the partition column value is showing up in the read result?

If not, can you add tests for this.

Reply:
updated the test with assertion to check the no. of split and value.